assets/js/util/test/numberFormat.js

```diff
@@ -115,5 +116,98 @@ describe( 'numberFormat', () => {
 	it.each( siteKitLocales )( 'formats numbers correctly with locale variant %s', ( locale, value, expected ) => {
 		setupGoogleSiteKit( locale );
 		expect( numberFormat( value ) ).toStrictEqual( expected );
+		expect( console ).not.toHaveWarned();
+	} );
+
+	const NumberFormat = Intl.NumberFormat;
+
+	/*
+	 * This function mocks the implementation of certain errors identified in
+	 * https://github.com/google/site-kit-wp/issues/3255
+	 * so that we can test that we've properly handled them.
+	 */
+	const mockNumberFormat = () => {
+		global.Intl.NumberFormat = ( locales, options ) => {
```

Comment:
This will probably work but it would be more appropriate to let Jest handle the mocking rather than doing this more manually. We should be able to do so as follows:

```js
NumberFormatSpy = jest.spyOn( global.Intl, 'NumberFormat' );
NumberFormatSpy.mockImplementation( ( locales, options ) => {
```

That way, we can clear the mock in an `afterEach` so it doesn't potentially cause other problems in the event of a failure. It also allows us to do things like `expect( global.Intl.NumberFormat ).toBeCalled( ... )` if we wanted to. The main benefits are consistency and resilient cleanup though.

Reply:
Updated in https://github.com/google/site-kit-wp/pull/3620/commits/98bec60f18fb859b20a62f0d978f98d2c5b30692